src/common/host_scalar_memory_storage.hpp

```diff
@@ -45,20 +47,32 @@ class host_scalar_memory_storage_t : public memory_storage_t {
 
     status_t get_scalar_value(void *value, size_t value_size) const {
         if (!is_initialized_ || value == nullptr
-                || value_size != types::data_type_size(data_type_))
+                || value_size != types::data_type_size(data_type_)
+                || value_size == 0 || value_size > max_scalar_size)
             return status::invalid_arguments;
 
-        std::memcpy(value, &scalar_storage_, value_size);
+        const auto *src
+                = reinterpret_cast<const std::uint8_t *>(&scalar_storage_);
+        auto *dst = static_cast<std::uint8_t *>(value);
+
+        std::copy(src, src + value_size, dst);
```

Comment:
Probably could use a comment what was the original version of the code and what's the reason for the current one for historical context.